tests/test_api.py

```diff
@@ -1281,6 +1285,57 @@ def test_submit_manifest_table_and_file_upsert(
         )
         assert response_csv.status_code == 200
 
+    @pytest.mark.synapse_credentials_needed
+    @pytest.mark.submission
+    def test_submit_nested_manifest_table_and_file_replace(
+        self,
+        client: FlaskClient,
+        request_headers: Dict[str, str],
+        helpers: Helpers,
+        synapse_store: SynapseStorage,
+    ) -> None:
+        # GIVEN the parameters to submit a manifest
+        params = {
+            "schema_url": DATA_MODEL_JSON_LD,
+            "data_type": "BulkRNA-seqAssay",
+            "restrict_rules": False,
+            "manifest_record_type": "table_and_file",
+            "asset_view": "syn63646213",
+            "dataset_id": "syn63646197",
+            "table_manipulation": "replace",
+            "data_model_labels": "class_label",
+            "table_column_names": "display_name",
+        }
+
+        # AND a test manifest with a nested file entity
+        test_upsert_manifest_path = helpers.get_data_path(
+            "mock_manifests/TestManifestOperation_test_submit_nested_manifest_table_and_file_replace.csv"
+        )
+
+        # AND a randomized annotation we can verify was added
+        df = pd.read_csv(test_upsert_manifest_path)
+        randomized_annotation_content = str(uuid.uuid4())
+        df["RandomizedAnnotation"] = randomized_annotation_content
+        csv_file = io.BytesIO()
+        df.to_csv(csv_file, index=False)
+        csv_file.seek(0)  # Rewind the buffer to the beginning
+
+        # WHEN I submit that manifest
+        response_csv = client.post(
+            "http://localhost:3001/v1/model/submit",
+            query_string=params,
+            data={"file_name": (csv_file, "test.csv")},
+            headers=request_headers,
+        )
+
+        # THEN the submission should be successful
+        assert response_csv.status_code == 200
+
+        # AND the file should be uploaded to Synapse with the new annotation
```

Comment:
> Do we also need to ensure that the manifest file on synapse gets cleaned up and the table gets deleted afterwards?


I do not believe so since as a part of the test the table needs to exist on Synapse with a static ID. As far as the manifest file being deleted, I do not think so either. I was adding the randomized annotation to verify that it went through the annotation update process; In conjunction with verifying the downloadAs attribute of the file should meet exactly what we need.

If the test modifies the name of the file, or does not update the annotation then we know that the logic was broken.

Reply:
> manifest file named synapse_storage_manifest_bulkrna-seqassay.csv gets added to the dataset folder

I added this logic.

> table was created

I am not sure what you mean for me to verify here. The assert_view has to be given in the params of the API, so by virtue of having a Synapse ID I can statically commit to source control the view exists already.

Reply:
Thanks @linglp I added this in the latest commit!